### ISSUE TITLE
Inverse battle mode

### ISSUE THREAD

**Issue by wavebeem** [wavebeem/pkmn.help#160]
An Inverse Battle (Japanese: さかさバトル Reverse Battle) is a [Pokémon battle](https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_battle) in the [Generation VI](https://bulbapedia.bulbagarden.net/wiki/Generation_VI) [core series](https://bulbapedia.bulbagarden.net/wiki/Core_series) games that were introduced in [Pokémon X and Y](https://bulbapedia.bulbagarden.net/wiki/Pok%C3%A9mon_X_and_Y). During an Inverse Battle, type matchups are reversed and no type is immune to any other type.

Any type that would be resistant or immune to another type is now weak to that type instead; also, any type that would be weak to another type is now resistant to that type instead.

**Comment by wavebeem** [comment]
Just got another request for this.

**Comment by wavebeem** [comment]
Oops, didn't mean to close it

**Comment by wavebeem** [comment]
And another request for this.

**Comment by wavebeem** [comment]
Another request

**Comment by wavebeem** [comment]
Another request.

**Comment by damocleas** [comment]
> Another request. Apparently this is a challenge mode you can enable in Pokérogue, and it's somewhat popular.

Came to see if this was already suggested, seems so! This would be another request, I use the tool a lot, very useful and efficient.
-PokéRogue Dev

**Comment by wavebeem** [comment]
@damocleas Thanks for the kind words 💚 If you wanted to work on this, I'd be happy to give you support over GitHub 🙏🏻 My life is pretty busy right now, so if this is something you'd like to see in the near future, that would probably help make this a reality 😅

**Comment by wavebeem** [comment]
If you do want to look into it (or for myself for future reference), the relevant file is `src/misc/data-matchups.ts`. a handful of functions are exported here which take `Generation` values. these will need to be updated to also take in some new parameter:

```ts
export const BattleMode = "regular" | "inverse";
```

and then we can modify the results so that 2 becomes 1/2, 1/2 becomes 2, and 0 becomes 1/2, or whatever the values are supposed to be.

Oh, and we would need to edit `src/screens/ScreenSettings.tsx` to actually allow selecting inverse mode.

or... and this is where I would need your help---is this something that maybe makes more sense to be toggled from within an offense/defense screen, because people will need to toggle it a lot?

**Comment by damocleas** [comment]
> If you do want to look into it (or for myself for future reference), the relevant file is `src/misc/data-matchups.ts`. a handful of functions are exported here which take `Generation` values. these will need to be updated to also take in some new parameter:
> 
> export const BattleMode = "regular" | "inverse";
> and then we can modify the results so that 2 becomes 1/2, 1/2 becomes 2, and 0 becomes 1/2, or whatever the values are supposed to be.
> 
> Oh, and we would need to edit `src/screens/ScreenSettings.tsx` to actually allow selecting inverse mode.
> 
> or... and this is where I would need your help---is this something that maybe makes more sense to be toggled from within an offense/defense screen, because people will need to toggle it a lot?

Feel like it could be a button to toggle below the rest as either some sort of "special modes" kind of thing or that but just as a general setting you have to change like the Gen type charts.

<img width="442" height="435" alt="Image" src="https://github.com/user-attachments/assets/1471d1d7-a160-4b28-9f8f-bfdd6d99425f" />

<img width="442" height="435" alt="Image" src="https://github.com/user-attachments/assets/d5d3d035-8b11-4875-a210-c6e2ce206835" />

Appreciate the quick response, will definitely look into it at some point.

**Comment by wavebeem** [comment]
Yeah, my preference would be putting alongside the generation type charts. It might be worth showing either some quick controls or a read-only overview of which settings have been selected (if they're not default), just so users don't forget they've enabled like... Gen 2-5 inverse battles or something lol.

**Comment by wavebeem** [comment]
@damocleas Would you mind taking a look at what I've started? I think it's good, but I'd love to have your help since I'm not an expert on this mode, and you're the one most recently interested in this:

PR: https://github.com/wavebeem/pkmn.help/pull/399

Test deployment: https://deploy-preview-399--pkmn-help.netlify.app/settings/

**Comment by wavebeem** [comment]
It looks right to me, so I'm just gonna deploy it. Please let me know if you have any feedback after using it.

**Comment by damocleas** [comment]
> [@damocleas](https://github.com/damocleas) Would you mind taking a look at what I've started? I think it's good, but I'd love to have your help since I'm not an expert on this mode, and you're the one most recently interested in this:
> 
> PR: [#399](https://github.com/wavebeem/pkmn.help/pull/399)
> 
> Test deployment: https://deploy-preview-399--pkmn-help.netlify.app/settings/

Oh! Didn't seem to get a notification for this, but I'll be sure to reply here though if there are any issues!
